### ISSUE TITLE
Error: fastlane failed. spawn fastlane ENOENT- VSTS Apple store deploy to Testflight

### ISSUE THREAD

**Issue by 199219921992** [Microsoft/app-store-vsts-extension#88]
I am getting Error: fastlane failed. spawn fastlane ENOENT error on using VSTS Apple store deploy task, I have installed fastlane on server and getting this issue.

Manually i have tried to push using command below and it worked but same the command when running from VSTS it fails with above error.

"fastlane pilot upload -u ******** -i /path/of /the/ipafile/test.ipa"

Any Quick assistance's appreciated.

**Comment by letko-dmitry** [comment]
Looks like the old one:
https://github.com/Microsoft/app-store-vsts-extension/issues/86

And still have no idea 😔

**Comment by 199219921992** [comment]
@letko-dmitry - Used fastlane command line to publish the IPA file instead of VSTS Task and went fine.

**Comment by letko-dmitry** [comment]
I decided not to use the plugin and switched to script and fastlane  )

**Comment by mrdaneeyul** [comment]
I had this issue, and when I switched to a command script task in VSTS (using the same command that the fastlane task was using), I got this error:

`2018-05-02T13:13:00.4923610Z /Users/admin/Documents/myagent/_work/_temp/4a285d67-ad1b-4afb-a19f-473b7ae3329d.sh: /usr/local/bin/fastlane: /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/bin/ruby: bad interpreter: No such file or directory
2018-05-02T13:13:00.5101820Z ##[error]Bash exited with code '126'.`

What worked for me was running the following commands on the Mac ([found here](https://github.com/CocoaPods/CocoaPods/issues/6898#issuecomment-333812742)):

`sudo gem update --system`
`sudo gem install -n /usr/local/bin fastlane`

`sudo gem update --system` threw an error, but it appears to have worked regardless for whatever reason.

Fixed it for the command line task, and turns out this fixed it for the fastlane task too, so I removed the command line task. Back in business. Hope this helps someone else.
